AndroidNativeKotlinTemplate/app/build.gradle

```diff
@@ -0,0 +1,40 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+dependencies {
+  compile project(':libs:SmartSync')
+  compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```

Comment:
Provides reflection support - which enables us to get the java class reference [here](https://github.com/wmathurin/SalesforceMobileSDK-Templates/blob/0547f48100a23e28240339c3db9eda005601db84/AndroidNativeKotlinTemplate/app/src/com/salesforce/androidnativekotlintemplate/MainApplication.kt#L42)